components/TextFields/src/MDCTextInputControllerLegacyDefault.m

```diff
@@ -0,0 +1,1261 @@
+    MDCTextInputDefaultFloatingPlaceholderScaleDefault;
+
+static UIColor *_activeColorDefault;
+static UIColor *_disabledColorDefault;
+static UIColor *_errorColorDefault;
+static UIColor *_floatingPlaceholderColorDefault;
+static UIColor *_inlinePlaceholderColorDefault;
+static UIColor *_normalColorDefault;
+
+static UITextFieldViewMode _underlineViewModeDefault = UITextFieldViewModeWhileEditing;
+
+@interface MDCTextInputControllerLegacyDefault () {
+  BOOL _mdc_adjustsFontForContentSizeCategory;
+
+  UIColor *_activeColor;
+  UIColor *_disabledColor;
+  UIColor *_floatingPlaceholderColor;
+  UIColor *_inlinePlaceholderColor;
+  UIColor *_normalColor;
+}
+
+@property(nonatomic, assign, readonly) BOOL isDisplayingCharacterCountError;
+@property(nonatomic, assign, readonly) BOOL isDisplayingErrorText;
+@property(nonatomic, assign, readonly) BOOL isPlaceholderUp;
+@property(nonatomic, assign) BOOL isRegisteredForKVO;
+
+@property(nonatomic, strong) MDCTextInputAllCharactersCounter *internalCharacterCounter;
+
+@property(nonatomic, strong) NSArray<NSLayoutConstraint *> *placeholderAnimationConstraints;
+
+@property(nonatomic, strong) NSLayoutConstraint *characterCountY;
+@property(nonatomic, strong) NSLayoutConstraint *characterCountTrailing;
+@property(nonatomic, strong) NSLayoutConstraint *clearButtonY;
+@property(nonatomic, strong) NSLayoutConstraint *clearButtonTrailingCharacterCountLeading;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderLeading;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderTop;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderTrailingCharacterCountLeading;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderTrailingSuperviewTrailing;
+
+@property(nonatomic, copy) NSString *errorAccessibilityValue;
+@property(nonatomic, copy, readwrite) NSString *errorText;
+@property(nonatomic, copy) NSString *previousLeadingText;
+
+@property(nonatomic, strong) UIColor *previousPlaceholderColor;
+
+@property(nonatomic, strong) UIFont *customLeadingFont;
+@property(nonatomic, strong) UIFont *customPlaceholderFont;
+@property(nonatomic, strong) UIFont *customTrailingFont;
+
+@end
+
+@implementation MDCTextInputControllerLegacyDefault
+
+@synthesize characterCounter = _characterCounter;
+@synthesize characterCountMax = _characterCountMax;
+@synthesize characterCountViewMode = _characterCountViewMode;
+@synthesize errorColor = _errorColor;
+@synthesize floatingPlaceholderScale = _floatingPlaceholderScale;
+@synthesize textInput = _textInput;
+@synthesize underlineViewMode = _underlineViewMode;
+
+// TODO: (larche): Support in-line auto complete.
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonMDCTextInputControllerDefaultInitialization];
+  }
+
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super init];
+  if (self) {
+    [self commonMDCTextInputControllerDefaultInitialization];
+
+    _activeColor =
```

Comment:
.activeColor actually handles nil in its setter and getting. It uses the default!